src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1869,6 +1869,21 @@ private void ImportCall(ILOpcode opcode, int token)
                 }
             }
 
+            // Hard coded InternalCall mappings for mono interoperability
+            if (callee.IsInternalCall)
+            {
+                var ecmaType = callee.OwningType as EcmaType;
```

Comment:
Nit: casting to `MetadataType` might be better - we don't need anything specific to `EcmaType`s here and `MetadataType` would also work if we end up needing this for things that are not backed by physical IL on disk (EcmaType represents just the types that exist in IL assemblies on disk - MetadataType is more general and includes things the compiler created out of thin air).

Reply:
thanks, done